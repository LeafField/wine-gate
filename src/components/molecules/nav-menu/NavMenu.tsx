import React, { FC } from "react";
import CategoryButton from "../../atoms/category-button/CategoryButton";
import FreeButton from "../../atoms/free-button/FreeButton";
import MenberButton from "../../atoms/menber-button/MenberButton";

const NavMenu: FC = () => {
  return (
    <div className="z-nav-menu fixed bottom-0 left-0 right-0 flex w-full bg-black/80 py-1 backdrop-blur-sm tablet:hidden">
      <MenberButton />
      <FreeButton />
      <CategoryButton />
    </div>
  );
};

export default NavMenu;
