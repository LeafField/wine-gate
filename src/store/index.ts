import { create } from "zustand";

type User = {
  username: string;
  id: string;
  email: string | undefined;
};

type Menu = "close" | "member" | "free" | "category";

type Store = {
  user: User | null;
  setUser: (user: User) => void;
  logout: () => void;
  menu: Menu;
  setMenu: (menu: Menu) => void;
};

export const useStore = create<Store>((set) => ({
  user: null,
  setUser: (user) => set({ user }),
  logout: () => set({ user: null }),
  menu: "close",
  setMenu: (menu) => set({ menu }),
}));
