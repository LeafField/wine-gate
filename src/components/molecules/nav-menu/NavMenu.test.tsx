import {
  screen,
  render,
  renderHook,
  act,
  waitFor,
} from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import NavMenu from "./NavMenu";
import { useStore } from "../../../store";

describe("NavMenuの結合テスト", () => {
  beforeAll(() => {
    const { result } = renderHook(() => useStore());
    act(() => {
      result.current.setMenu("close");
    });
  });

  test("初期状態では会員メニュー、フリーワード検索、カテゴリー検索のボタンが表示されている", () => {
    const { container } = render(<NavMenu />);
    expect(
      screen.getByRole("button", { name: "会員メニュー" }),
    ).toBeInTheDocument();
    expect(
      screen.getByRole("button", { name: "フリーワード検索" }),
    ).toBeInTheDocument();
    expect(
      screen.getByRole("button", { name: "カテゴリー検索" }),
    ).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test("会員メニューのボタンをクリックすると、会員メニューのボタンが閉じるボタンに変わる", async () => {
    expect.assertions(3);
    render(<NavMenu />);
    await userEvent.click(screen.getByRole("button", { name: "会員メニュー" }));
    await waitFor(() => {
      expect(screen.getByText("閉じる")).toBeInTheDocument();
      expect(screen.getByText("フリーワード検索")).toBeInTheDocument();
      expect(screen.getByText("カテゴリー検索")).toBeInTheDocument();
    });
  });

  test("閉じるボタンをクリックすると、会員メニューのボタンに変わる", async () => {
    render(<NavMenu />);
    await userEvent.click(screen.getByRole("button", { name: "閉じる" }));
    await waitFor(() => {
      expect(
        screen.getByRole("button", { name: "会員メニュー" }),
      ).toBeInTheDocument();
    });
  });

  test("フリーワード検索ボタンをクリックすると、閉じるボタンに変わる", async () => {
    render(<NavMenu />);
    await userEvent.click(
      screen.getByRole("button", { name: "フリーワード検索" }),
    );
    return expect(screen.getByText("閉じる")).toBeInTheDocument();
  });

  test("閉じるボタンをクリックすると、フリーワード検索ボタンに変わる", async () => {
    render(<NavMenu />);
    await userEvent.click(screen.getByRole("button", { name: "閉じる" }));
    await waitFor(() => {
      expect(
        screen.getByRole("button", { name: "フリーワード検索" }),
      ).toBeInTheDocument();
    });
  });
});
