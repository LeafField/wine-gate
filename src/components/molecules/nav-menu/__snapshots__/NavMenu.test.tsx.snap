// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavMenuの結合テスト 初期状態では会員メニュー、フリーワード検索、カテゴリー検索のボタンが表示されている 1`] = `
<div>
  <div
    class="z-nav-menu fixed bottom-0 left-0 right-0 flex w-full bg-black/80 py-1 backdrop-blur-sm tablet:hidden"
  >
    <button
      class="flex w-full flex-col items-center justify-center border-r border-white"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-user text-menu-icon text-white"
        data-icon="user"
        data-prefix="far"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M304 128a80 80 0 1 0 -160 0 80 80 0 1 0 160 0zM96 128a128 128 0 1 1 256 0A128 128 0 1 1 96 128zM49.3 464H398.7c-8.9-63.3-63.3-112-129-112H178.3c-65.7 0-120.1 48.7-129 112zM0 482.3C0 383.8 79.8 304 178.3 304h91.4C368.2 304 448 383.8 448 482.3c0 16.4-13.3 29.7-29.7 29.7H29.7C13.3 512 0 498.7 0 482.3z"
          fill="currentColor"
        />
      </svg>
      <span
        class="text-menu-title text-white"
      >
        会員メニュー
      </span>
    </button>
    <button
      class="flex w-full flex-col items-center justify-center border-r border-white"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-magnifying-glass text-menu-icon text-white"
        data-icon="magnifying-glass"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"
          fill="currentColor"
        />
      </svg>
      <span
        class="text-menu-title text-white"
      >
        フリーワード検索
      </span>
    </button>
    <button
      class="flex w-full flex-col items-center justify-center"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-list text-menu-icon text-white"
        data-icon="list"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M40 48C26.7 48 16 58.7 16 72v48c0 13.3 10.7 24 24 24H88c13.3 0 24-10.7 24-24V72c0-13.3-10.7-24-24-24H40zM192 64c-17.7 0-32 14.3-32 32s14.3 32 32 32H480c17.7 0 32-14.3 32-32s-14.3-32-32-32H192zm0 160c-17.7 0-32 14.3-32 32s14.3 32 32 32H480c17.7 0 32-14.3 32-32s-14.3-32-32-32H192zm0 160c-17.7 0-32 14.3-32 32s14.3 32 32 32H480c17.7 0 32-14.3 32-32s-14.3-32-32-32H192zM16 232v48c0 13.3 10.7 24 24 24H88c13.3 0 24-10.7 24-24V232c0-13.3-10.7-24-24-24H40c-13.3 0-24 10.7-24 24zM40 368c-13.3 0-24 10.7-24 24v48c0 13.3 10.7 24 24 24H88c13.3 0 24-10.7 24-24V392c0-13.3-10.7-24-24-24H40z"
          fill="currentColor"
        />
      </svg>
      <span
        class="text-menu-title text-white"
      >
        カテゴリー検索
      </span>
    </button>
  </div>
</div>
`;
